src/librustdoc/html/static/js/search.js

```diff
@@ -404,17 +405,17 @@ window.initSearch = function(rawSearchIndex) {
                 // a levenshtein distance value that isn't *this* good so it goes
                 // into the search results but not too high.
                 lev_distance = Math.ceil((checkGenerics(obj, val) + lev_distance) / 2);
-            } else if (obj.length > GENERICS_DATA && obj[GENERICS_DATA].length > 0) {
+            }
+            if (obj.length > GENERICS_DATA && obj[GENERICS_DATA].length > 0) {
```

Comment:
Same here. Why is this necessary? I'm not comfortable reviewing changes to the JS search code other than adding `[NAME]` as necessary.

Reply:
Because before, if we found that the item had a "close enough" name to what we're looking for, it then ran checks on generics as well, otherwise it was going to look into the generics.

Now, in both cases we go look into generics and then we pick the best match out of the two.

And yes, sorry for the JS changes... If that can make you feel better, you can always confirm it's working as expected by running the JS search tests.

Reply:
The way I interpret your description is that it's not necessary to bundle the JS change with the `html/formats/cache.rs` change; it's a separate bugfix. If so, can you please move this change to a separate PR and assign someone who's more knowledgeable about `search.js` on it? Or ask them to review the JS part of this PR?

> And yes, sorry for the JS changes... If that can make you feel better, you can always confirm it's working as expected by running the JS search tests.

Rustdoc doesn't currently have very exhaustive test coverage, so I'd rather have someone more knowledgeable look over the JS change.